api/v1/connection_types.go

```diff
@@ -89,6 +89,21 @@ type ConnectionPostgres struct {
 	InsecureTLS bool         `json:"insecureTLS,omitempty"`
 }
 
+type ConnectionPrometheus struct {
+	// URL is the connection url.
+	URL types.EnvVar `json:"url,omitempty"`
+
+	Username     types.EnvVar      `json:"username,omitempty"`
+	Password     types.EnvVar      `json:"password,omitempty"`
+	ClientID     types.EnvVar      `json:"clientID,omitempty"`
+	ClientSecret types.EnvVar      `json:"clientSecret,omitempty"`
+	Scopes       []string          `json:"scope,omitempty" yaml:"scope,omitempty"`
+	TokenURL     string            `json:"tokenURL,omitempty" yaml:"tokenURL,omitempty"`
+	Params       map[string]string `json:"params,omitempty" yaml:"params,omitempty"`
```

Comment:
Can we move this under an oauth struct ?

Reply:
👍🏽